base/meta.jl

```diff
@@ -223,6 +223,56 @@ function parse(str::AbstractString; raise::Bool=true, depwarn::Bool=true)
     return ex
 end
 
+"""
+    parsefile(filename; raise=true, depwarn=true)
+
+Parse the Julia file located at `filename`.
+Returns a vector that contains all expressions in the file.
+Throws an `IOError` if the file `filename` does not exist.
+If `raise` is `true` (default),
+syntax errors will raise an error; otherwise, `parse` will return an expression that will
+raise an error upon evaluation.  If `depwarn` is `false`, deprecation warnings will be
+suppressed.
+
+```julia
+julia> Meta.parsefile("foo.jl")
+```
+"""
+function parsefile(filename::AbstractString;
+                   raise::Bool=true,
+                   depwarn::Bool=true)::Vector{Union{Expr, LineNumberNode}}
+    if !ispath(filename)
+        throw(IOError("The file \"$(filename)\" does not exist"))
+    end
+    full_file_path::String = realpath(filename)::String
+    file_contents::String = read(full_file_path,
+                                 String)::String
+    block::Expr = Meta.parse("""
+                             begin
+                                 $(file_contents)
+                             end
+                             """;
+                             raise = raise,
+                             depwarn = depwarn)::Expr
+    block_fixed::Expr = _parsefile_fix_filename(block,
+                                                full_file_path)::Expr
+    lines::Vector{Union{Expr, LineNumberNode}} = convert(Vector{Union{Expr, LineNumberNode}}, block_fixed.args)::Vector{Union{Expr, LineNumberNode}}
```

Comment:
Just a question: this type is defined/asserted three times. Is there a nice way of not doing that?

Reply:
Yeah I think I can remove one of those.

Reply:
Take a look now.

We have to, for example, keep the `convert(TYPE, something)::TYPE`. The compiler is not able to infer the return type of `convert`.